build_projects/dotnet-host-build/PublishTargets.cs

```diff
@@ -70,9 +73,10 @@ public static BuildTargetResult FinalizeBuild(BuildTargetContext c)
             if (CheckIfAllBuildsHavePublished())
             {
                 string targetContainer = $"{Channel}/Binaries/Latest/";
-                string targetVersionFile = $"{targetContainer}{SharedFrameworkNugetVersion}";
+                string targetVersionFile = $"{targetContainer}{BranchName}-{CommitCount}";
```

Comment:
per offline, let's just replace this with SHA of the commit. It encompasses CommitCount and disambiguates across branches.